pyiron_base/dataclasses/job.py

```diff
@@ -0,0 +1,12 @@
+from dataclasses import dataclass
+from typing import List, Optional
+
+
+@dataclass
+class Executable:
+    version: Optional[str]
+    name: str
+    operation_system_nt: bool
+    executable: Optional[str]
+    mpi: bool
+    accepted_return_codes: List[int]
```

Comment:
```suggestion
    name: str
    operation_system_nt: bool
    mpi: bool
    accepted_return_codes: List[int]
    version: Optional[str] = None
    executable: Optional[str] = None
```

optional attributes must be given a default and specified last, otherwise construction of the dataclass still requires them.